test/integration/user-management-api/applications_test.rb

```diff
@@ -65,7 +66,7 @@ def setup
         get admin_api_applications_path, params: params.merge({ service_id: @service.id })
 
         assert_response :success
-        assert_equal 2, Nokogiri::XML::Document.parse(@response.body).xpath("//applications").children.length
+        assert_equal @token.owner.accessible_cinstances.size, Nokogiri::XML::Document.parse(@response.body).xpath("//applications").children.length
```

Comment:
I actually liked `2` more because it tests something that we know, not something that might be broken in the same way in the ruby expression and the api call. :)

Reply:
I prefer the other version, 2 is like a magic constant, if you create more applications then this test fails but it should not :)

Reply:
The way I see it, this test is about "getting applications that are accessible by the buyer". We test the integration between user permission (token) and the endpoint.
Say we changed the user permissions, made it more strict, and now the buyer can only access 1 of the 2 applications. If the endpoint logic is outdated or broken, the test will pass, because we have a hardcoded 2 and nothing to tell us this 2 is no longer correct.
WDYT?